react/week5/github-usersearch/src/userProfile.js

```diff
@@ -0,0 +1,37 @@
+import React, { useState, useEffect } from "react";
+import { Link, useParams } from "react-router-dom";
+
+function Profile() {
+  const { username } = useParams();
+  const [item, setItem] = useState({});
+
+    useEffect(() => {
+      let isFetched = true
+      fetch(`https://api.github.com/users/${username}`)
+      .then(response => response.json())
+      .then(item => {
+        if(isFetched) {
+          setItem(item);
+        }
+      })
+      return () => isFetched = false
+    })
```

Comment:
useEffect is a bit weird. I think I understand why you need the `isFetched` boolean, however, I would consider using a dependancy array to control when useEffect is call instead
```suggestion
    useEffect(() => {
      fetch(`https://api.github.com/users/${username}`)
      .then(response => response.json())
      .then(item => {
          setItem(item);
      })
    }, [username])
```

This code basically says that useEffect should only run if `username` changes. The docs are pretty dense on useEffect: https://reactjs.org/docs/hooks-effect.html

I'd recommend going through them when you have time. There is some good stuff in there.